api/app/db/crud/hfi_calc.py

```diff
@@ -145,6 +145,29 @@ def store_hfi_station(session: Session, station_code: int, fuel_type_id: int, pl
     session.commit()
 
 
+def toggle_ready(session: Session,
+                 fire_centre_id: int,
+                 planning_area_id: int,
+                 date_range: DateRange,
+                 username: str) -> HFIReady:
+    """ Toggles the planning area ready state for an hfi request """
+    now = get_utc_now()
+    hfi_request = get_most_recent_updated_hfi_request(session, fire_centre_id, date_range)
+    ready_state: HFIReady = session.query(HFIReady)\
+        .filter(HFIReady.planning_area_id == planning_area_id)\
+        .filter(HFIReady.hfi_request_id == hfi_request.id)\
+        .first()
+    if ready_state.ready is True:
+        ready_state.ready = False
+    else:
+        ready_state.ready = True
```

Comment:
couldn't you replace these last 4 lines with just `ready_state.ready = !ready_state.ready` ?

Reply:
Yeah nice call. I can do that in a follow up PR.